qtconsole/frontend_widget.py

```diff
@@ -233,14 +229,18 @@ def _is_complete(self, source, interactive):
             prompt created. When triggered by an Enter/Return key press,
             'interactive' is True; otherwise, it is False.
         """
-        self._input_splitter.reset()
-        try:
-            complete = self._input_splitter.push(source)
-        except SyntaxError:
-            return True
-        if interactive:
-            complete = not self._input_splitter.push_accepts_more()
-        return complete
+        kc = self.blocking_client
+        msg_id = kc.is_complete(source)
+        while True:
+            try:
+                reply = kc.shell_channel.get_msg(block=True, timeout=self.is_complete_timeout)
```

Comment:
Just to check my understanding, because this is a completely separate client, pulling these messages and discarding them if they're not the one we want won't influence the main asynchronous client that the Qt console is using, right?